### ISSUE TITLE
unable indicate a characteristic

### ISSUE THREAD

**Issue by ThuanNH4** [dariuszseweryn/RxAndroidBle#795]
Hello everyone, thank you for the great job with this library 🎉

I used this lib for my app but I facing the following problem:
`2022-06-16 17:23:04.378 24292-25487/com.polidea.rxandroidble.sample E/AndroidRuntime: FATAL EXCEPTION: RxBleThread-1
Process: com.polidea.rxandroidble.sample, PID: 24292
io.reactivex.exceptions.UndeliverableException: The exception could not be delivered to the consumer because it has already canceled/disposed the flow or the exception has nowhere to go to begin with. Further reading: [https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#error-handling](https://github.com/ReactiveX/RxJava/wiki/What%27s-different-in-2.0#error-handling) | com.polidea.rxandroidble2.exceptions.BleCannotSetCharacteristicNotificationException: Cannot write client characteristic config descriptor (code 3) with characteristic UUID 7dab2102-2f7c-403d-9e7e-4ba3731b2daf
at io.reactivex.plugins.RxJavaPlugins.onError(RxJavaPlugins.java:367)
at io.reactivex.subjects.PublishSubject$PublishDisposable.onError(PublishSubject.java:314)
at io.reactivex.subjects.PublishSubject.onError(PublishSubject.java:243)`

The above log was thrown when I call the method:
<img width="693" alt="Screen Shot 2022-06-16 at 23 38 51" src="https://user-images.githubusercontent.com/60633595/174123181-ba901160-6092-4511-ba34-98b43905a4d8.png">

I also tried indicate my characteristic with NRF (https://play.google.com/store/apps/details?id=no.nordicsemi.android.mcp&hl=en&gl=US) and it works fine
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/60633595/174124740-8e06d2f0-1dac-4bf9-a272-48d8b9f9ccb2.png)

Can you check and help me to resolve this issue? 
Thank!

**Comment by dariuszseweryn** [comment]
For UndeliverableException check wiki

**Comment by ThuanNH4** [comment]
Hi, thanks for reply but this only handle the exception (UndeliverableException), the indication feature still doesn't work. My app need this feature to work
please help. thank you

**Comment by ThuanNH4** [comment]
before migrate to your lib (RxAndroidBLE), in my app I used BluetoothGatt to enable indication and it works fine for me.
![image](https://user-images.githubusercontent.com/60633595/174540580-0f5b2a64-abf9-4794-a562-be5f03901732.png)
so I think this is an issue of RxAndroidBLE

**Comment by ThuanNH4** [comment]
ok let me try again, will attach log when done

**Comment by ThuanNH4** [comment]
hi @dariuszseweryn 
below is log on my app for both 2 case
before migrate to RxAndroidBLE
```
2022-06-21 13:42:35.426 9041-9041/app.test V/AudioManager: querySoundEffectsEnabled...
2022-06-21 13:42:35.443 9041-9041/app.test D/BaseFragment: A_Setup_07_Fragment onAttach
2022-06-21 13:42:35.444 9041-9041/app.test D/BaseFragment: A_Setup_07_Fragment onCreate
2022-06-21 13:42:35.473 9041-9041/app.test D/BaseFragment: A_Setup_07_Fragment onViewCreated
2022-06-21 13:42:35.495 9041-9041/app.test I/AppName#SharedPreferencesHelper: getWifiName null
2022-06-21 13:42:35.496 9041-9041/app.test D/___LOG___: A_Setup_07_Fragment onResume
2022-06-21 13:42:35.497 9041-9041/app.test D/___LOG___: stt blt = false
2022-06-21 13:42:35.497 9041-9041/app.test D/___LOG___: stt Per = false
2022-06-21 13:42:35.499 9041-9041/app.test D/___LOG___: Start flow: Bluetooth Wifi List Search Flow.
2022-06-21 13:42:35.512 9041-9041/app.test I/HwViewRootImpl: removeInvalidNode all the node in jank list is out of time
2022-06-21 13:42:37.503 9041-9041/app.test D/___LOG___: write data to char 7dab2101-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 13:42:37.507 9041-9041/app.test D/___LOG___: Write 7dab2101-2f7c-403d-9e7e-4ba3731b2daf successfully
2022-06-21 13:42:37.509 9041-9041/app.test D/___LOG___: indicate characteristic 7dab2102-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 13:42:38.024 9041-9041/app.test D/___LOG___: Indicate 7dab2102-2f7c-403d-9e7e-4ba3731b2daf: true
2022-06-21 13:42:38.024 9041-9041/app.test D/___LOG___: Subcription Successfully
2022-06-21 13:42:39.547 9041-9041/app.test D/___LOG___: action intent + com.example.bluetooth.le.ACTION_DATA_AVAILABLE
2022-06-21 13:42:39.549 9041-9041/app.test D/___LOG___: value: 01 from: 7dab2102-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 13:42:39.551 9041-9041/app.test D/___LOG___: read characteris 7dab2103-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 13:42:39.554 9041-9041/app.test D/___LOG___: read Successfully
2022-06-21 13:42:39.637 9041-9041/app.test D/___LOG___: action intent + com.example.bluetooth.le.ACTION_DATA_AVAILABLE
2022-06-21 13:42:39.638 9041-9041/app.test D/___LOG___: value: 0C00 from: 7dab2103-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 13:42:39.643 9041-9041/app.test D/___LOG___: write data to char 7dab2104-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 13:42:39.645 9041-9041/app.test D/___LOG___: Write 7dab2104-2f7c-403d-9e7e-4ba3731b2daf successfully
2022-06-21 13:42:39.646 9041-9041/app.test D/___LOG___: read2105
2022-06-21 13:42:39.647 9041-9041/app.test D/___LOG___: read characteris 7dab2105-2f7c-403d-9e7e-4ba3731b2daf
```

after migrate to RxAndroidBLE
```
2022-06-21 12:46:51.788 2784-2784/lixil.leopard.intg V/AudioManager: querySoundEffectsEnabled...
2022-06-21 12:46:51.832 2784-2784/lixil.leopard.intg I/HwViewRootImpl: removeInvalidNode all the node in jank list is out of time
2022-06-21 12:46:51.865 2784-3006/lixil.leopard.intg I/RxBle#ConnectionOperationQueue: RUNNING CharacteristicWriteOperation{MAC='38:B8:00:13:8C:5C', characteristic=[uuid='7dab2101-2f7c-403d-9e7e-4ba3731b2daf', hexValue=[31]]}
2022-06-21 12:46:51.978 2784-2864/lixil.leopard.intg I/RxBle#GattCallback: MAC='38:B8:00:13:8C:5C' onCharacteristicWrite(), status=0, value=[uuid='7dab2101-2f7c-403d-9e7e-4ba3731b2daf']
2022-06-21 12:46:51.982 2784-2784/lixil.leopard.intg D/___LEOPARD___: Write success to UUID: 7dab2101-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 12:46:51.983 2784-3006/lixil.leopard.intg I/RxBle#ConnectionOperationQueue: RUNNING DescriptorWriteOperation{MAC='38:B8:00:13:8C:5C', descriptor=[uuid='00002902-0000-1000-8000-00805f9b34fb', hexValue=[02, 00]]}
2022-06-21 12:46:52.125 2784-2864/lixil.leopard.intg I/RxBle#GattCallback: MAC='38:B8:00:13:8C:5C' onDescriptorWrite(), status=3, value=[uuid='00002902-0000-1000-8000-00805f9b34fb']
2022-06-21 12:46:52.142 2784-2784/lixil.leopard.intg D/___LEOPARD___: Error: com.polidea.rxandroidble2.exceptions.BleCannotSetCharacteristicNotificationException: Cannot write client characteristic config descriptor (code 3) with characteristic UUID 7dab2102-2f7c-403d-9e7e-4ba3731b2daf
2022-06-21 12:46:52.143 2784-3006/lixil.leopard.intg I/RxBle#ConnectionOperationQueue: RUNNING DescriptorWriteOperation{MAC='38:B8:00:13:8C:5C', descriptor=[uuid='00002902-0000-1000-8000-00805f9b34fb', hexValue=[00, 00]]}
2022-06-21 12:46:52.225 2784-2864/lixil.leopard.intg I/RxBle#GattCallback: MAC='38:B8:00:13:8C:5C' onDescriptorWrite(), status=3, value=[uuid='00002902-0000-1000-8000-00805f9b34fb'
```
you can see the indicator 2102 works fine for me but when I switch to RxBLE it doesn't work anymore

**Comment by dariuszseweryn** [comment]
> before migrate to your lib (RxAndroidBLE), in my app I used BluetoothGatt to enable indication and it works fine for me.
> ![image](https://github.com/assets/60633595/0f5b2a64-abf9-4794-a562-be5f03901732)
> so I think this is an issue of RxAndroidBLE

The log from before the migration does not show anything related to result of descriptor write unfortunately nor it does match the code you pasted. It is hard to relate to that.
In the logs you can clearly see that RxAndroidBle does receive status=3 from the peripheral so the library does behave as expected. If you want to ignore it you may use a `NotificationSetupMode.COMPAT` and do the descriptor write yourself and ignore the result.

**Comment by dariuszseweryn** [comment]
Closing due to lack of information and inactivity of the original poster.

When more data will be available—feel free to add that information and reopen this issue.

Best Regards
